tests/handlers/filesystem/test_squashfs.py

```diff
@@ -0,0 +1,184 @@
+00000060  00 00 00 00 00 00 cb 00  00 00 00 00 00 01 14 00  |................|
+00000070  00 00 00 00 00 01 34 78  da 4b 2c 28 c8 49 35 e4  |......4x.K,(.I5.|
+00000080  4a 04 51 46 10 ca 18 42  99 70 01 00 8b ee 09 3b  |J.QF...B.p.....;|
+00000090  00 39 78 da 53 5c c2 f0  3f 89 79 f3 45 06 06 06  |.9x.S\\..?.y.E...|
+000000a0  66 06 4c c0 ae 88 90 67  41 97 43 93 67 45 93 e7  |f.L....gA.C.gE..|
+000000b0  43 93 67 42 93 17 05 c9  0b fe 05 cb 5f 07 b2 19  |C.gB........_...|
+000000c0  c1 6a 38 e1 f2 6c 00 6f  a6 13 17 00 35 78 da 63  |.j8..l.o....5x.c|
+000000d0  66 80 00 66 06 26 4e 06  86 c4 82 82 9c 54 43 bd  |f..f.&N......TC.|
+000000e0  92 8a 12 46 20 97 11 cc  35 02 71 99 80 5c 26 30  |...F ...5.q..\\&0|
+000000f0  d7 18 c4 65 66 e2 fc ff  1f cc 35 01 71 01 cb 90  |...ef.....5.q...|
+00000100  11 84 80 10 00 00 00 00  00 00 00 77 00 00 00 19  |...........w....|
+00000110  00 00 00 00 00 00 00 00  00 00 01 02 00 16 78 da  |..............x.|
+00000120  63 60 00 83 06 08 c5 90  c0 80 0a 14 a0 b4 03 00  |c`..............|
+00000130  1b 68 01 41 00 00 00 00  00 00 01 1c 00 00 00 00  |.h.A............|
+00000140
+"""
+)
+
+
+def pad_contents(contents: bytes, pad_len: int):
+    return contents + b"\0" * (pad_len - len(contents))
+
+
+@pytest.mark.parametrize(
+    "contents, handler_class",
+    (
+        pytest.param(SQUASHFS_V4_NO_PAD_CONTENTS, SquashFSv4Handler, id="v4"),
+        pytest.param(SQUASHFS_V3_LE_NO_PAD_CONTENTS, SquashFSv3Handler, id="v3_le"),
+        pytest.param(SQUASHFS_V3_BE_NO_PAD_CONTENTS, SquashFSv3Handler, id="v3_be"),
+    ),
+)
+@pytest.mark.parametrize(
+    "start",
+    (
+        pytest.param(b"", id="zero_start"),
+        pytest.param(b"A" * 128, id="non_zero_start"),
+    ),
+)
+@pytest.mark.parametrize(
+    "pad_len",
+    (
+        pytest.param(None, id="no_pad"),
+        pytest.param(1024, id="1k_pad"),
+        pytest.param(4096, id="4k_pad"),
+    ),
+)
+@pytest.mark.parametrize(
+    "extra",
+    (
+        pytest.param(
+            b"",
+            id="no_extra_end",
+        ),
+        pytest.param(
+            b"A" * 128,
+            id="extra_end",
+        ),
+        pytest.param(
+            b"A" * 4096,
+            id="extra_long_end",
+        ),
+        pytest.param(
+            b"\0" * 128,
+            id="extra_null_end",
+        ),
+        pytest.param(
+            b"\0" * 128 + b"A" + b"\0" * 4096,
+            id="extra-non-null-pad",
+        ),
+    ),
+)
+def test_squashfs_chunk_is_detected(
+    contents: bytes, handler_class, start: bytes, pad_len: Optional[int], extra: bytes
+):
+    start_offset = len(start)
+    if pad_len is None:
+        pad_len = len(contents)
+    complete_content = pad_contents(contents, pad_len)
+
+    chunk = handler_class().calculate_chunk(
+        io.BytesIO(start + complete_content + extra), start_offset
+    )
+
+    assert chunk.start_offset == start_offset
+    assert chunk.end_offset == start_offset + pad_len
```

Comment:
If you agree with the comments above, this could work:

```suggestion
def test_squashfs_chunk_is_detected(
    contents: bytes, handler_class, start: bytes, alignment: Optional[int], extra: bytes
):
    start_offset = len(start)
    if alignment is None:
        alignment = 1
    complete_content = pad_contents(contents, alignment)

    chunk = handler_class().calculate_chunk(
        io.BytesIO(start + complete_content + extra), start_offset
    )
    padded_size = round_up(len(contents), alignment)
    assert chunk.start_offset == start_offset
    assert chunk.end_offset == start_offset + padded_size
```